Chapter10/pretty_print.erl

```diff
@@ -136,59 +127,51 @@ pad(_, Word) -> Word.
 
 
 
-
-%%Compresser (Removes duplicates) [1,1,1,2,2,5,7,9,9] -> [1,2,5,7,9]
-compress([]) -> [];
-compress(List) -> compress(List, []).
-
-compress([], Output) -> Output;
-compress(List, Output) when length(List) == 1 -> Output ++ List;	
-compress([Head|Tail], _) when (Tail - Head) =< 1 -> [];
-compress([Head|Tail], Output) when (Tail - Head) >= 1 -> Output ++ [Head];
-compress([Head,Next|Tail], Output) ->
-	if
-		(Next - Head) == 0 -> compress([Head|Tail], Output);
-		% (Next - Head) == 1 -> compress([Next|Tail], Output ++ [Head] ++ ["-"] );
-		(Next - Head) >= 1 -> compress([Next|Tail], Output ++ [Head])
-	end.
-
-
 %%Linker (Links adjacent numbers) [1,2,5,8,9,10] -> [{1,2},{5},{8,10}]
 compress_link([]) -> [];
 compress_link(List) -> compress_link(List, {0, 0}, []).
 
-compress_link([], _, Output) -> Output;
-compress_link([Head,Next], Range, Output) -> 
-	if
-		(Next - Head == 1) and (Range == {0, 0}) -> Output ++ [{Head, Next}];
-		(Next - Head == 1) -> Output ++ [{element(1, Range), Next}];
-		(Next - Head >= 1) and (Range /= {0, 0}) -> Output ++ [{element(1, Range),Head},{Next}];
-		(Next - Head >= 1) -> Output ++ [{Head},{Next}]
-	end;
-compress_link([Head,Next|Tail], Range, Output) ->
-	if
-		(Next - Head == 1) and (Range == {0, 0}) -> compress_link([Next|Tail], {Head, 0}, Output);
-		(Next - Head == 1) -> compress_link([Next|Tail], {element(1, Range), Next}, Output);
-		(Next - Head >= 1) and (Range == {0, 0}) -> compress_link([Next|Tail], {0, 0}, Output ++ [{Head}]);
-		(Next - Head >= 1) -> compress_link([Next|Tail], {0, 0}, Output ++ [{element(1, Range),Head}])
-	end.
-
-
-%Prettyer (List with tuples to string) [{1,2},{3},{5,9}] -> "1-2,3,5-9,"
-compress_pretty([]) -> [];
-compress_pretty([Head|Tail]) when tuple_size(Head) == 1 -> char_to_integer(element(1,Head)) ++ "," ++ compress_pretty(Tail);
-compress_pretty([Head|Tail]) when tuple_size(Head) == 2 ->
-	char_to_integer(  element(1,Head)) ++ "-" ++ char_to_integer(element(2,Head)  )
-	++ "," ++ compress_pretty(Tail).
-
-
-char_to_integer(Char) -> lists:flatten(io_lib:format("~p", [Char])).
-
-
+compress_link([], _Range, Output) -> Output;
 
+compress_link([Head,Next], {0, 0}, Output)
+  when (Next - Head == 1) -> Output ++ [{Head, Next}];
```

Comment:
the same comment regarding `Acc ++ NewElement`. This is not efficient, see previous PRs for link to efficiency concern. If you want to change the order, use `lists:reverse/1` once processing has finished, it should be more efficient.

Reply:
Changed the code to merge [ | ] then reverse once complete